### ISSUE TITLE
When an affinity isn't "matched", swarm returns an error of constraint

### ISSUE THREAD

**Issue by aanm** [docker/swarm#977]
```bash
sudo docker -H 127.0.0.1:2373 run \                                                                                                                                                                                       
 -e "constraint:node==node1" \                                                                                                                                                                                                               
 -e "affinity:image==busybox:latest" \                                                                                                                                                                                                       
 busybox \                                                                                                                                                                                                                                   
 echo hello world                                                                                                                                                                                                                            
FATA[0000] Error response from daemon: unable to find a node that satisfies node==node1
```
None of the nodes has the image `busybox:latest` but the message should be more clear.

If I remove the constraint I get
```bash
sudo docker -H 127.0.0.1:2373 run \
-e "affinity:image==busybox:latest" \
busybox \
echo hello world
FATA[0000] Error response from daemon: unable to find a node that satisfies image==busybox:latest
```

**Comment by mbentley** [comment]
I've also come across this when trying to use labels for nodes when the images don't exist since it has the misleading `unable to find a node that satisfies label==value`.

**Comment by nishanttotla** [comment]
@aanm @mbentley this should be fixed in #1796, can you please try the latest RC (1.2.0-rc1) and let us know if your issue was resolved?

**Comment by vieux** [comment]
@aanm @mbentley I'm going to close this one since we believe it was fixed in #1796, feel free to comment here if it's not fixed.
